pkg/pillar/zedUpload/commonutils/types.go

```diff
@@ -0,0 +1,33 @@
+// Copyright(c) 2017-2018 Zededa, Inc.
+// All rights reserved.
+
+package commonutils
+
+import (
+	"fmt"
+)
+
+// ChunkData contains the details of Chunks being downloaded
+type ChunkData struct {
+	Size  int64  // complete size to upload/download
```

Comment:
Any particular reason why you didn't use uint64 here? A size can't be negative unless -1 is used to signal an error.
And uint64 means you can avoid some explicit type conversions.

Reply:
the default value type of file size returned from s3 and azure is int64, so I aligned it with int64.

Reply:
ok